### PR TITLE
feat: upgrade to Palm

### PR DESCRIPTION
**Pull request by jfavellar90**
This PR aims to apply the changes required to migrate the tutor-notes plugin to Palm

### CODE REVIEW

File: tutornotes/plugin.py [hunk 1]
```diff
@@ -26,18 +29,27 @@
...
-))
+
+# To add a custom initialization task, create a bash script template under:
+# tutorcodejail/templates/codejail/tasks/
+# and then add it to the MY_INIT_TASKS list. Each task is in the format:
+# ("<service>", ("<path>", "<to>", "<script>", "<template>"))
+MY_INIT_TASKS: list[tuple[str, tuple[str, ...], int]] = [
+    ("mysql", ("notes", "tasks", "mysql", "init"), tutor_hooks.priorities.HIGH),
+    ("lms", ("notes", "tasks", "lms", "init"), tutor_hooks.priorities.HIGH),
+    ("notes", ("notes", "tasks", "notes", "init"), tutor_hooks.priorities.HIGH),
```
Comment (regisb) [inline tutornotes/plugin.py:40]: Why are these tasks high priority? Unless there is a good reason to do so, we should add them as normal priority.

File: tutornotes/plugin.py [hunk 2]
```diff
@@ -104,3 +116,12 @@ def _mount_edx_notes_api(volumes, name):
     ]
 )
 tutor_hooks.Filters.CONFIG_OVERRIDES.add_items(list(config.get("overrides", {}).items()))
+
+# Notes public hosts
+@tutor_hooks.Filters.APP_PUBLIC_HOSTS.add()
+def _notes_public_hosts(hosts: list[str], context_name: t.Literal["local", "dev"]) -> list[str]:
+    if context_name == "dev":
+        hosts += ["notes.{{ LMS_HOST }}:8000"]
```
Comment (regisb) [inline tutornotes/plugin.py:124]: Port should be 8120, not 8000.

**Review by regisb — CHANGES_REQUESTED** [review]
Thanks for the PR Jhony!

File: tutornotes/plugin.py (hunk 2 above)
Comment (jfavellar90) [reply tutornotes/plugin.py]: Addressed

**Comment by jfavellar90** [comment]
@regisb this is ready to go, take a look if you want

File: tutornotes/plugin.py [hunk 3]
```diff
@@ -103,4 +116,15 @@ def _mount_edx_notes_api(volumes, name):
...
-tutor_hooks.Filters.CONFIG_OVERRIDES.add_items(list(config.get("overrides", {}).items()))
+tutor_hooks.Filters.CONFIG_OVERRIDES.add_items(
+    list(config.get("overrides", {}).items())
+)
+
+# Notes public hosts
+@tutor_hooks.Filters.APP_PUBLIC_HOSTS.add()
+def _notes_public_hosts(hosts: list[str], context_name: t.Literal["local", "dev"]) -> list[str]:
+    if context_name == "dev":
+        hosts += ["notes.{{ LMS_HOST }}:8120"]
```
Comment (regisb) [inline tutornotes/plugin.py:127]: This should be `{{ NOTES_HOST }}:8120` (and same below)

**Review by regisb — CHANGES_REQUESTED** [review]
Almost ready!

File: tutornotes/plugin.py (hunk 3 above)
Comment (jfavellar90) [reply tutornotes/plugin.py]: That's a nice catch. Fixed

**Comment by regisb** [comment]
(Same comment as for the xqueue plugin)

Could you modify the docker-compose config to remove the notes-permissions container? Instead, in Palm there is now a single "permissions" container that makes all permission changes. See this commit for reference: https://github.com/overhangio/tutor/pull/827/commits/bfa1f657282d53f8a9c1820ebcb27d630836ac58

**Pull request merged**